File: backend/backend/schemas/public_apply.py
"""Public (no-auth) schemas for Phase 6, Sub-phase 6C — Flow B, public link."""
from pydantic import BaseModel, EmailStr
from typing import Optional
from uuid import UUID

from backend.schemas.public_invitations import RedeemedSessionInfo


class PublicApplyContext(BaseModel):
    """What a candidate sees on the public-apply landing page."""
    job_title: str
    job_description: Optional[str] = None
    seniority: Optional[str] = None
    candidate_instructions: Optional[str] = None
    duration_minutes: int


class PublicRegisterRequest(BaseModel):
    name: str
    email: EmailStr
    resume_id: Optional[UUID] = None


class PublicRegisterResponse(BaseModel):
    access_token: str
    token_type: str = "bearer"
    session: RedeemedSessionInfo
    livekit_token: str
    livekit_url: str

File: frontend/src/services/api/publicApply.ts
import { fetchApi } from "../../lib/api";
import type { RedeemedSessionInfo } from "./publicInvitations";

export interface PublicApplyContext {
  job_title: string;
  job_description?: string;
  seniority?: string;
  candidate_instructions?: string;
  duration_minutes: number;
}

export interface PublicRegisterResponse {
  access_token: string;
  token_type: string;
  session: RedeemedSessionInfo;
  livekit_token: string;
  livekit_url: string;
}

export async function getApplyContext(token: string): Promise<PublicApplyContext> {
  return fetchApi<PublicApplyContext>(`/api/v1/apply/${token}`);
}

export async function registerApplicant(
  token: string,
  data: { name: string; email: string }
): Promise<PublicRegisterResponse> {
  return fetchApi<PublicRegisterResponse>(`/api/v1/apply/${token}/register`, {
    method: "POST",
    data,
  });
}
